coordinator/src/routes/health/get-status.ts

```diff
@@ -11,21 +12,45 @@ export interface StatusCheckResponse {
   links?: string
 }
 
-export async function serviceHealthCheck(url: string): Promise<StatusCheckResponse> {
-  return await axios
-    .get<string>(url, {timeout: 20000})
-    .then((response): StatusCheckResponse => ({
+export async function serviceHealthCheck(url: string, logger: pino.Logger): Promise<StatusCheckResponse> {
+  try {
+    const response = await axios.get<string>(url, {timeout: 20000})
+    return {
       status: response.status === 200 ? "pass" : "error",
       timeout: "false",
       responseCode: response.status,
       outcome: response.data,
       links: url
-    }))
-    .catch(error => ({
+    }
+  } catch (error) {
+    logger.error(error)
+    return {
       status: "error",
       timeout: error.code === "ECONNABORTED" ? "true" : "false",
       responseCode: 500
-    }))
+    }
+  }
+}
+
+function createStatusResponse(checks: { [p: string]: Array<StatusCheckResponse> }, h: Hapi.ResponseToolkit) {
```

Comment:
I think the type `{ [p: string]: Array<StatusCheckResponse> }` can also be represented as `Record<String, Array<StatusCheckResponse>>`. Not sure whether that's nicer or not :shrug:

Reply:
done